core/src/test/scala/unit/kafka/coordinator/group/GroupMetadataManagerTest.scala

```diff
@@ -610,6 +610,51 @@ class GroupMetadataManagerTest {
   }
 
   @Test
+  def testLoadGroupAndOffsetsFromDifferentSegments(): Unit = {
```

Comment:
That seems debatable since the cleaner will kick in for older segments. In any case, if the underlying issue is more general, we should probably update the JIRA and PR titles.